presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3ClientFactory.java

```diff
@@ -85,77 +91,55 @@ synchronized AmazonS3 getS3Client(Configuration config, HiveClientConfig clientC
             s3UserAgentSuffix = "presto-select";
         }
 
-        ClientConfiguration clientConfiguration = new ClientConfiguration()
-                .withMaxErrorRetry(maxErrorRetries)
-                .withProtocol(sslEnabled ? Protocol.HTTPS : Protocol.HTTP)
-                .withConnectionTimeout(toIntExact(connectTimeout.toMillis()))
-                .withSocketTimeout(toIntExact(socketTimeout.toMillis()))
-                .withMaxConnections(maxConnections)
-                .withUserAgentPrefix(userAgentPrefix)
-                .withUserAgentSuffix(s3UserAgentSuffix);
-
-        AWSCredentialsProvider awsCredentialsProvider = getAwsCredentialsProvider(config, defaults);
-        AmazonS3Builder<? extends AmazonS3Builder, ? extends AmazonS3> clientBuilder = AmazonS3Client.builder()
-                .withCredentials(awsCredentialsProvider)
-                .withClientConfiguration(clientConfiguration)
-                .withMetricsCollector(new PrestoS3FileSystemMetricCollector(PrestoS3FileSystem.getFileSystemStats()))
-                .enablePathStyleAccess();
-
-        boolean regionOrEndpointSet = false;
-
-        String endpoint = config.get(S3_ENDPOINT);
-        boolean pinS3ClientToCurrentRegion = config.getBoolean(S3_PIN_CLIENT_TO_CURRENT_REGION, defaults.isPinS3ClientToCurrentRegion());
-        verify(!pinS3ClientToCurrentRegion || endpoint == null,
-                "Invalid configuration: either endpoint can be set or S3 client can be pinned to the current region");
-
-        // use local region when running inside of EC2
-        if (pinS3ClientToCurrentRegion) {
-            Region region = Regions.getCurrentRegion();
-            if (region != null) {
-                clientBuilder.withRegion(region.getName());
-                regionOrEndpointSet = true;
-            }
-        }
-
-        if (!isNullOrEmpty(endpoint)) {
-            clientBuilder.withEndpointConfiguration(new EndpointConfiguration(endpoint, null));
-            regionOrEndpointSet = true;
-        }
-
-        if (!regionOrEndpointSet) {
-            clientBuilder.withRegion(US_EAST_1);
-            clientBuilder.setForceGlobalBucketAccessEnabled(true);
-        }
-
-        s3Client = clientBuilder.build();
-        return s3Client;
+        AwsCredentialsProvider awsCredentialsProvider = getAwsCredentialsProvider(config, defaults);
+
+        ClientOverrideConfiguration clientOverrideConfiguration = ClientOverrideConfiguration.builder()
+                .retryStrategy(retryStrategy -> retryStrategy.maxAttempts(maxErrorRetries))
+                .apiCallTimeout(ofMillis(socketTimeout.toMillis()))
+                .apiCallAttemptTimeout(ofMillis(connectTimeout.toMillis()))
```

Comment:
nit: we don't need this

The correct equivalent for `withConnectionTimeout` is `httpClientBuilder.connectionTimeout`, which we are already setting below.